server/controllers/stock/inventory/core.js

```diff
@@ -38,11 +39,58 @@ var errors = {
 exports.getIds = getIds;
 exports.getItemsMetadata = getItemsMetadata;
 exports.getItemsMetadataById = getItemsMetadataById;
+exports.createItemsMetadata = createItemsMetadata;
+exports.updateItemsMetadata = updateItemsMetadata;
 exports.hasBoth = hasBoth;
 exports.errors = errors;
 exports.errorHandler = errorHandler;
 
 /**
+* Create inventory metadata in the database
+*
+* @function createItemsMetadata
+* @return {Promise} Returns a database query promise
+*/
+function createItemsMetadata(req, next) {
+  let record = req.body;
+
+  record.enterprise_id = req.session.enterprise.id;
+  record.uuid = db.bid(record.uuid || uuid.v4());
+  record.group_uuid = db.bid(record.group_uuid);
+
+  let sql = `INSERT INTO inventory SET ?;`;
+  return db.exec(sql, [record])
+  .then(() => {
+    return uuid.unparse(record.uuid);
+  })
+  .catch(next);
+}
+
+/**
+* Update inventory metadata in the database
+*
+* @function updateItemsMetadata
+* @return {Promise} Returns a database query promise
+*/
+function updateItemsMetadata(req, next) {
+  let record = req.body;
+  let identifier = db.bid(req.params.uuid);
+
+  record.uuid = identifier;
+  record.group_uuid = db.bid(record.group_uuid);
+
+  let sql = `UPDATE inventory SET ? WHERE uuid = ?;`;
+  return db.exec(sql, [record, identifier])
+  .then(() => {
```

Comment:
Just so that you know, with fat arrow syntax, you can write this line:
```js
.then(() => {
  return getItemsMetadataById(identifier);
})
```
... like this ...
```js
.then(() => getItemsMetadataById(identifier));
```
The `return` keyword is implicit if it is the only statement in the body of the "fat arrow" function.